hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/heartbeat/HoodieHeartbeatClient.java

```diff
@@ -250,11 +250,8 @@ public boolean isHeartbeatExpired(String instantTime) throws IOException {
   }
 
   private void updateHeartbeat(String instantTime) throws HoodieHeartbeatException {
-    try {
+    try (OutputStream outputStream = this.fs.create(new Path(heartbeatFolderPath + Path.SEPARATOR + instantTime), true)) {
       Long newHeartbeatTime = System.currentTimeMillis();
-      OutputStream outputStream =
-          this.fs.create(new Path(heartbeatFolderPath + Path.SEPARATOR + instantTime), true);
-      outputStream.close();
```

Comment:
It has been closed in this line?

Reply:
OutputStream may not close gracefully. I found it is similar to https://github.com/apache/hudi/pull/9309

Reply:
If we want to flush something or see the heartbeats immediately, we need to close the outputstream before we do the heartbeat check. Do you think it is necessary?

Reply:
ok, I understant. No need to do this.